src/core/space.jl

```diff
@@ -374,27 +374,52 @@ Return an agent given its ID.
 """
 id2agent(id::Integer, model::ABM) = model.agents[id]
 
-# TODO: Rename `node_neighbors` to `space_neighbors`, and
-# extend it for continuous space with third argument "radius"
-
 """
-    node_neighbors(agent::AbstractAgent, model::ABM)
-    node_neighbors(node::Int, model::ABM)
+    space_neighbors(position, model::ABM [, r]) → ids
+
+Return the ids of the agents neighboring the given `position` (which must match type
+with the spatial structure of the `model`). If `r` is given, it is the radius to search
+for agents.
+
+For `DiscreteSpace`s `r` must be integer and defines higher degree neighbors.
+For example, for `r=2` include first and second degree neighbors,
+that is, neighbors and neighbors of neighbors.
+
+For `ContinuousSpace`, `r` is real number and finds all neighbors within distance `r`
```

Comment:
I think we should add another column to the database: diameter. When searching for neighbors, we should check whether diameter of neighbors is within r. Currently, it checks the center coordinates.

Reply:
this is a good point, but I'd prefer to discuss it in a separate issue / pr. It seems to be a strict improvement of the existing functionality, but its good to have the basic in first.

Reply:
@kavir1698 can you please open an _issue_ for this idea so we won]'t forget it and have a place to think about it?